torch/csrc/api/include/torch/nn/modules/loss.h

```diff
@@ -117,5 +117,31 @@ struct TORCH_API CosineEmbeddingLossImpl : Module {
 /// PyTorch's module storage semantics.
 TORCH_MODULE(CosineEmbeddingLoss);
 
+// ============================================================================
+
+/// Creates a criterion that optimizes a multi-label one-versus-all
+/// loss based on max-entropy, between input :math:`x` and target :math:`y` of size
+/// :math:`(N, C)`.
+struct TORCH_API MultiLabelSoftMarginLossImpl : Module {
```

Comment:
My sincere apologies for missing this earlier: for all `torch::nn` layers we will need to subclass it from `Cloneable` in the following way:
```suggestion
struct TORCH_API MultiLabelSoftMarginLossImpl : : public torch::nn::Cloneable<MultiLabelSoftMarginLossImpl> {
```
otherwise `module->clone()` won't work on them. I think for all currently open PRs we should change this, and I will fix the modules that are already in the codebase (e.g. L1Loss).

Reply:
Okay, no problem! Thanks for the review!